src/Plugins/PluginAutoload.php

```diff
@@ -133,24 +141,143 @@ protected function findPluginBaseDir($path)
         return $this->findPluginBaseDir($checkDir);
     }
 
+    /**
+     * Return the path to the autoload file, relative to the
+     * provided path, if it exists.
+     */
+    protected function checkAutoloadPath($path)
+    {
+        $autoloadFile = "$path/vendor/autoload.php";
+        if (file_exists($autoloadFile)) {
+            return $autoloadFile;
+        }
+    }
+
+    /**
+     * Compare the contents of the composer.json and composer.lock files
+     * of the plugin being used, and see if anything contaied therein is
+     * incompatible with the projects loaded by Terminus itself.
+     */
     protected function validateComposerLock($pluginBaseDir, $terminusSrcDir)
     {
+        $pluginComposerJsonFile = $pluginBaseDir . '/composer.json';
+        $pluginComposerLockFile = $pluginBaseDir . '/composer.lock';
+        $pluginComposerJson = $this->loadJson($pluginComposerJsonFile);
+        $pluginComposerLock = $this->loadJson($pluginComposerLockFile);
+
         // If there is no composer.lock file, that means that
         // the plugin has autoload classes, but requires no dependencies.
-        if (!file_exists("$pluginBaseDir/composer.lock")) {
+        // In this case, we know it is safe to load the autoload file.
+        if (empty($pluginComposerLock)) {
             return;
         }
 
-        // TODO: Load the composer.lock and analyze it against
-        // dirname($terminusSrcDir) . '/composer.lock'.
-        throw new TerminusException("Autoloading plugin dependencies is not supported yet.");
+        $terminusComposerJsonFile = dirname($terminusSrcDir) . '/composer.json';
+        $terminusComposerLockFile = dirname($terminusSrcDir) . '/composer.lock';
```

Comment:
My general understanding is that `/` is okay for use with PHP APIs, but DIRECTORY_SEPARATOR is needed when shelling out, or comparing against paths that come from the system, etc.; however, I saw some folks claiming that DIRECTORY_SEPARATOR use was more widely necessary than that. Since I'm not an expert on PHP + Windows, I'll go ahead and use DIRECTORY_SEPARATOR here just to be conservative.